core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -109,10 +109,6 @@ class HadoopRDD[K, V](
     minPartitions: Int)
   extends RDD[(K, V)](sc, Nil) with Logging {
 
-  if (initLocalJobConfFuncOpt.isDefined) {
-    sparkContext.clean(initLocalJobConfFuncOpt.get)
```

Comment:
technically this might also regress behavior since it's `HadoopRDD` is not private, but it's probably OK if this change improves performance since this class is a `DeveloperApi`.